packages/terra-application/tests/jest/application-loading-overlay/ApplicationLoadingOverlay.test.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import uuidv4 from 'uuid/v4';
+
+import ApplicationLoadingOverlay from '../../../src/application-loading-overlay/ApplicationLoadingOverlay';
+import ApplicationLoadingOverlayContext from '../../../src/application-loading-overlay/ApplicationLoadingOverlayContext';
+
+jest.mock('uuid/v4');
+
+describe('ApplicationLoadingOverlay', () => {
+  let reactUseContext;
+  let loadingOverlayContextValue;
+
+  beforeAll(() => {
+    uuidv4.mockReturnValue('test-id');
+
+    /**
+     * Until Enzyme is updated with full support for hooks, we need to
+     * mock out the useContext implementation.
+     */
+    reactUseContext = React.useContext;
+    React.useContext = (contextValue) => {
+      if (ApplicationLoadingOverlayContext === contextValue) {
+        return loadingOverlayContextValue;
+      }
+      return reactUseContext(contextValue);
+    };
+  });
+
+  afterAll(() => {
+    React.useContext = reactUseContext;
+  });
+
+  it('should render loading overlay as closed', () => {
+    loadingOverlayContextValue = {
```

Comment:
why don't we define this globally and return it in the beforeAll useContext mock, instead of setting it per it block?

Reply:
ah, we need to reset it - should we do a reset in the afterall then?